gitpod-io/spring-petclinic
# Bring latest commits from upstream project and delete the .idea folder

## Description


Two issues were causing the project to fail to start up with Maven Wrapper correctly detected:
- We were using an outdated version of Maven Wrapper
- Committing anything in the .idea folder will cause IntelliJ to not auto-configure Maven.
    - We tried leaving the `.idea/runConfigurations` folder in the repo, for example.

I've tried doing one or another, but the only way the project worked correctly with Maven Wrapper was when I applied both changes.

As we noticed the project was working fine with the [upstream Spring-PetClinic](https://github.com/spring-projects/spring-petclinic), we decided to bring the updates that were missing in this repository.

The run configuration was left under `.run` folder due to this [Blog Post from JetBrains](https://blog.jetbrains.com/idea/2020/03/intellij-idea-2020-1-beta2/):
> Note that if you’ve added the `.idea` folder to `.gitignore`, the IDE will suggest you store your run configurations in the `…/project_folder/.run/` directory by default.

ℹ️  This PR should not be squashed upon merge; otherwise, we'll lose the history, making it difficult to sync with the [upstream repository](https://github.com/spring-projects/spring-petclinic) later.

## Repository root files comparison

| [Before](https://github.com/gitpod-io/spring-petclinic/tree/f55e8a2dfcd0f97c3405f7a20fc1a1cfeb8f16fd) | [After](https://github.com/gitpod-io/spring-petclinic/tree/vn/update-from-upstream) |
| --- | --- |
| <img width="893" alt="image" src="https://user-images.githubusercontent.com/418083/191777079-27925b7d-60a2-4bbc-b040-7f9d04fd6195.png"> |  <img width="891" alt="image" src="https://user-images.githubusercontent.com/418083/191926912-a164a057-feaa-4e33-b8bc-a525e92911c1.png"> |

## Note about a known issue

In IntelliJ, sometimes you may see this warning, but it's harmless and has already been scheduled for fix ([IDEA-300044](https://youtrack.jetbrains.com/issue/IDEA-300044/Opening-new-project-gets-warning-Cannot-resolve-Maven-home-Use-Maven-wrapper-is-not-correct-maven-home-reverting-to-embedded)), so you can ignore it:

<img width="510" alt="image" src="https://user-images.githubusercontent.com/418083/191781710-8baeb3a3-b7b8-4596-a6a7-271b5d3048d8.png">

## How to test

1. Open a workspace, on <gitpod.io>, with IntelliJ IDEA selected as preferred IDE, from the branch of this PR: https://gitpod.io/#https://github.com/gitpod-io/spring-petclinic/tree/vn/update-from-upstream
    a. Or open it from JetBrains Gateway:
    <img width="790" alt="image" src="https://user-images.githubusercontent.com/418083/191732823-c21f81ea-600a-4f3c-a312-6d6bc76ac5dd.png">
2. As it was [prebuilt](https://github.com/gitpod-io/spring-petclinic/pull/67#issuecomment-1254903974), confirm if the project starts fast (i.e. it doesn't need to download the dependencies before starting the PetClinic server), and if "petclinic" is correctly identified as a Maven project, on the sidebar:
    <img width="176" alt="image" src="https://user-images.githubusercontent.com/418083/191735615-48788144-3891-4986-bb57-4edff95d1032.png">
3. Open these two workspaces in VS code, to confirm they are rendering the same website:
    - https://gitpod.io/#https://github.com/spring-projects/spring-petclinic
    - https://gitpod.io/#https://github.com/gitpod-io/spring-petclinic/tree/vn/update-from-upstream
    <img width="268" alt="image" src="https://user-images.githubusercontent.com/418083/191773266-0ff6d3ea-d6f7-4fe1-af62-88b9cd9bf3ff.png">
4. Confirm if clicking the Debug Button, on 'Run Configuration' menu, allows debugging the app properly.
    <img width="1624" alt="image" src="https://user-images.githubusercontent.com/418083/191916238-412e70e9-cbe6-46a5-bad3-43e9880b7457.png">
